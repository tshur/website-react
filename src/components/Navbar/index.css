:local .nav {
  background: green;
  height: 100%;
  width: 70px;
  float: left;
  overflow: hidden;
}

:local .button_spacer {
  background: yellow;
  margin-top: 70px;
  text-align: center;
}

:local .button {
  background: yellow;
  position: relative;
  margin-top: 70px;
  height: 120px;
  width: 70px;
  text-align: center;
}

:local .link {
  text-decoration: none;
}

:local .text {
  position: absolute;
  top: 50%;
  left: 50%;
  font-size: 24px;

  transform: translateX(-50%) translateY(-50%) rotate(-90deg);
}