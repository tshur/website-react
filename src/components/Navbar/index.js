import React from 'react';
import { Route, Link, Switch } from 'react-router-dom';
import styles from './index.css';

import Body from '../Body';
import NotFound from '../NotFound';

const Navbar = () =>
  <div className='container'>
    <div className={styles.nav}>
      <ul>
        <li>
          <Link to='/'> Home </Link>
        </li>
        <li>
          <Link to='/about'> About </Link>
        </li>
        <li>
          <Link to='/projects'> Projects </Link>
        </li>
      </ul>
    </div>

    <div className={styles.content}>
      <Switch>
        <Route exact path='/' component={Body} />
        <Route path='/about' component={Body} />
        <Route path='/projects' component={Body} />
        <Route component={NotFound} />
      </Switch>
    </div>
  </div>

export default Navbar;