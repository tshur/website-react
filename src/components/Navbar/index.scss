.nav {
  background: green;
  height: 100%;
  width: 70px;
  float: left;
  overflow: hidden;

  .button {
    background: yellow;
    position: relative;
    margin-top: 70px;
    height: 120px;
    width: 70px;
    text-align: center;

    .text {
      position: absolute;
      top: 50%;
      left: 50%;
      font-size: 24px;
      color: black;

      transform: translateX(-50%) translateY(-50%) rotate(-90deg);
    }
  }
}
