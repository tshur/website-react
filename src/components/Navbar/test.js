import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import Navbar from '.';

describe('Navbar', () => {

  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter><Navbar /></BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
