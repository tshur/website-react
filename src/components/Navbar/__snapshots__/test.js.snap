// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar has a valid snapshot 1`] = `
<div
  className="container undefined"
>
  <a
    className="link"
    href="/"
    onClick={[Function]}
  >
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        Home
      </div>
    </div>
  </a>
  <a
    className="link"
    href="/about"
    onClick={[Function]}
  >
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        About
      </div>
    </div>
  </a>
  <a
    className="link"
    href="/projects"
    onClick={[Function]}
  >
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        Projects
      </div>
    </div>
  </a>
</div>
`;
